03week/ticTacToe.js

```diff
@@ -23,32 +23,82 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
+
 function horizontalWin() {
-  // Your code here
-}
+  for (var r=0; r<3; r++) {
+    let line = board[r];
+    if ((line[0] === line[1] && line[1] === line[2]) && (line[0] === 'X' || line[0] === 'O')) {
+      return true;
+      }
+    }
+  };
+
 
 function verticalWin() {
-  // Your code here
-}
+  for (var c=0; c<3; c++) {
+    let line = [];
+    for (var r=0; r<3; r++){
+      line.push(board[r][c]);
+    }if ((line[0] === line[1] && line[1] === line[2]) && (line[0] === 'X' || line[0] === 'O')) {
+      return true;
+      }
+    }
+  };
 
 function diagonalWin() {
-  // Your code here
-}
+  if ((board[0][0] === 'X' || board[0][0] === 'O') && ((board[0][0] === board[1][1]) && (board[1][1] === board[2][2]))) {
+    return true;
+  }else if ((board[0][2] === 'X' || board[0][2] === 'O') && ((board[0][2] === board[1][1]) && (board[1][1] === board[2][0]))) {
+    return true;
+  }
+};
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin() == true) {
+    return true;
+  }else if (verticalWin() == true) {
+    return true;
+  }else if (diagonalWin() == true) {
+    return true;
+  }
+}
+function turns(){
+  let sum = 0;
```

Comment:
nice, i like this approach to figuring out how many turns have passed. 
A different approach might be to keep a counter that increments each time you process a valid turn.